File: JavaHTTPRecipesEdition1/com/heatonresearch/httprecipes/ch12/recipe1/LoadRSS.java
package com.heatonresearch.httprecipes.ch12.recipe1;

import java.io.*;
import java.net.*;

import javax.xml.parsers.*;

import org.xml.sax.*;

import com.heatonresearch.httprecipes.rss.*;

/**
 * Recipe #12.1: Display an RSS Feed
 * Copyright 2007 by Jeff Heaton(jeff@jeffheaton.com)
 *
 * HTTP Programming Recipes for Java Bots
 * ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 *
 * This recipe shows how to use the RSS class to load a
 * RSS feed.  The RSS feed is then displayed.
 *
 * This software is copyrighted. You may use it in programs
 * of your own, without restriction, but you may not
 * publish the source code without the author's permission.
 * For more information on distributing this code, please
 * visit:
 *    http://www.heatonresearch.com/hr_legal.php
 *
 * @author Jeff Heaton
 * @version 1.1
 */
public class LoadRSS
{

  /**
   * Display an RSS feed.
   * @param url The URL of the RSS feed.
   * @throws IOException Thrown if an IO exception occurs.
   * @throws SAXException Thrown if an XML parse exception occurs.
   * @throws ParserConfigurationException Thrown if there is an error setting
   * up the parser.
   */
  public void process(URL url) throws IOException, SAXException,
      ParserConfigurationException
  {
    RSS rss = new RSS();
    rss.load(url);
    System.out.println(rss.toString());
  }

  /**
   * The main method processes the command line arguments and
   * then calls process to display the feed.
   * 
   * @param args Holds the RSS feed to download.
   */
  public static void main(String args[])
  {
    try
    {
      URL url;

      if (args.length != 0)
        url = new URL(args[0]);
      else
        url = new URL("http://www.httprecipes.com/1/12/rss1.xml");

      LoadRSS load = new LoadRSS();
      load.process(url);
    } catch (Exception e)
    {
      e.printStackTrace();
    }
  }
}

File: CSharpHTTPRecipesEdition1/HTTPRecipes/Recipe12_1/LoadRSS.cs
using System;
using System.Collections.Generic;
using System.Text;
using HeatonResearch.Spider.RSS;

namespace Recipe12_1
{
    /// <summary>
    /// Recipe #12.1: Display an RSS Feed
    /// Copyright 2007 by Jeff Heaton(jeff@jeffheaton.com)
    ///
    /// HTTP Programming Recipes for C# Bots
    /// ISBN: 0-9773206-7-7
    /// http://www.heatonresearch.com/articles/series/20/
    ///
    /// This recipe shows how to use the RSS class to load a
    /// RSS feed.  The RSS feed is then displayed.
    ///
    /// This software is copyrighted. You may use it in programs
    /// of your own, without restriction, but you may not
    /// publish the source code without the author's permission.
    /// For more information on distributing this code, please
    /// visit:
    ///    http://www.heatonresearch.com/hr_legal.php
    /// </summary>
    class LoadRSS
    {        
        /// <summary>
        /// Display an RSS feed.
        /// </summary>
        /// <param name="url">The URL of the RSS feed.</param>
        public void Process(Uri url)
        {
            RSS rss = new RSS();
            rss.Load(url);
            Console.WriteLine(rss.ToString());
        }

        static void Main(string[] args)
        {
            Uri url;

            if (args.Length != 0)
                url = new Uri(args[0]);
            else
                url = new Uri("http://www.httprecipes.com/1/12/rss1.xml");

            LoadRSS load = new LoadRSS();
            load.Process(url);
        }
    }
}
